quinn-proto/src/connection/mod.rs

```diff
@@ -1895,15 +1901,12 @@ impl Connection {
 
         let was_closed = self.state.is_closed();
         let was_drained = self.state.is_drained();
-        let stateless_reset = self
-            .peer_params
-            .stateless_reset_token
-            .map_or(false, |token| {
-                packet.payload.len() >= RESET_TOKEN_SIZE
-                    && packet.payload[packet.payload.len() - RESET_TOKEN_SIZE..] == token[..]
-            });
 
-        let result = match self.decrypt_packet(now, &mut packet) {
+        let decrypted = packet.map_or(Err(None::<TransportError>), |mut packet| {
+            let number = self.decrypt_packet(now, &mut packet)?;
+            Ok((packet, number))
+        });
```

Comment:
I think this might be easier to follow as an explicit `match`?

Reply:
Done.